artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/buffer/TimedBuffer.java

```diff
@@ -93,10 +82,10 @@
    public TimedBuffer(final int size, final int timeout, final boolean logRates) {
       bufferSize = size;
 
-      this.logRates = logRates;
-
       if (logRates) {
-         logRatesTimer = new Timer(true);
+         this.flushProfiler = Profiler.instrumented();
+      } else {
+         this.flushProfiler = Profiler.none();
```

Comment:
would it not be better to set to null, as so don't then need a Noop implementation, and later do a guard statement so the code can be compiled out if not enabled..